dips/dip-160.md

```diff
@@ -0,0 +1,59 @@
+---
+dip: 160
+title: Off-chain API ping command
+authors: Xiao Li (@xli)
+status: Draft
+discussions-to: https://github.com/diem/dip/issues/160
+type: Informational
+created: 04/08/2021
+requires: DIP-1
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Summary
+
+This DIP describes an off-chain API v2 ping command that makes it easier for clients to detect network and protocol problems.
+
+# Abstract
+
+To improve the ability of off-chain service handling network problems and protocol errors, we’d like to introduce a ping command for client to:
+1. detect dead connections to a remote off-chain service.
+2. measure the latency to a remote off-chain service.
+3. detect off-chain protocol errors to a remote off-chain service, for example: when a remote off-chain service rotates their compliance key, and client used old compliance public key to verify the response message, the ping command can be used to debug the error and verify the fix.
```

Comment:
maybe simplify this last one. For example, an off-chain service responds back with an unexpected compliance key due to key rotation, this can be leveraged to debug and verify the inconsistency.